x-pack/plugins/security_solution/server/lib/detection_engine/signals/threat_mapping/utils.ts

```diff
@@ -113,3 +114,28 @@ export const combineConcurrentResults = (
 
   return combineResults(currentResult, maxedNewResult);
 };
+
+const separator = '___SEPARATOR___';
+export const encodeThreatMatchNamedQuery = ({
+  id,
+  field,
+  value,
+}: ThreatMatchNamedQuery): string => {
+  return [id, field, value].join(separator);
```

Comment:
This is the crux of this enrichment feature: we encode these values into a string to identify the given query filter, and then decode these values on the response in order to build the enrichment.